nbformat/tests/base.py

```diff
@@ -7,13 +7,13 @@
 
 import os
 import unittest
-
+import io
 
 class TestsBase(unittest.TestCase):
     """Base tests class."""
 
-    def fopen(self, f, mode=u'r'):
-        return open(os.path.join(self._get_files_path(), f), mode)
+    def fopen(self, f, mode=u'r',encoding='utf-8'):
+        return io.open(os.path.join(self._get_files_path(), f), mode, encoding=encoding)
```

Comment:
This will fail if anything asks for a binary mode, because you don't specify an encoding to read a file in binary mode:

```
 3│ f=  open('Dockerfile', 'rb', encoding='utf-8')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-e8ab8a38bf40> in <module>()
----> 1 f=  open('Dockerfile', 'rb', encoding='utf-8')

ValueError: binary mode doesn't take an encoding argument
```